### PR TITLE
Issue 1656: Upgrade spring-webmvc to version 5.2.3

### LINKED ISSUE #1656: Upgrade org.springframework:spring-webmvc to version 5.2.3 or higher
# Task Description

We should upgrade `org.springframework:spring-webmvc` to version `5.2.3` or higher, in order to fix a cross-site request forgery (as explained [here](https://snyk.io/vuln/SNYK-JAVA-ORGSPRINGFRAMEWORK-542933)) and a reflected file download vulnerability (as explained [here](https://snyk.io/vuln/SNYK-JAVA-ORGSPRINGFRAMEWORK-542935)).

The dependency upgrade should be done in the [strongbox-parent](https://github.com/strongbox/strongbox-parent/) project.

# Acceptance Test

* [ ] Building the code with `mvn clean install -Dintegration.tests` still works.
* [ ] Running `mvn spring-boot:run` in the `strongbox-web-core` still starts up the application correctly.
* [ ] Building the code and running the `strongbox-distribution` from a `zip` or `tar.gz` still works.
* [ ] The tests in the [`strongbox-web-integration-tests`](https://github.com/strongbox/strongbox-web-integration-tests/) still run properly.

# Help

* [Our chat](https://chat.carlspring.org/)
* Points of contact:
  * @carlspring
  * @fuss86
  * @sbespalov
  * @steve-todorov

### PR DESCRIPTION
**Pull request by savaliyagautam**
# Pull Request Description

This pull request closes #1656 

Fixed test cases `NugetArtifactControllerTest`, `MavenArtifactControllerTest` and `LayoutRequestMappingTest`

# Acceptance Test

* [x] Building the code with `mvn clean install -Dintegration.tests` still works.
* [x] Running `mvn spring-boot:run` in the `strongbox-web-core` still starts up the application correctly.
* [x] Building the code and running the `strongbox-distribution` from a `zip` or `tar.gz` still works.
* [x] The tests in the [`strongbox-web-integration-tests`](https://github.com/strongbox/strongbox-web-integration-tests/) still run properly.

# Questions

* Does this pull request break backward compatibility? 
  * [ ] Yes
  * [x] No

* Does this pull request require other pull requests to be merged first? 
  * [x] Yes, please see https://github.com/strongbox/strongbox-parent/pull/61
  * [ ] No

* Does this require an update of the documentation?
  * [ ] Yes, please see strongbox/strongbox-docs#{PR_NUMBER}
  * [x] No

### CODE REVIEW

File: strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/layout/nuget/NugetArtifactControllerTest.java [hunk 1]
```diff
@@ -369,17 +373,16 @@ public void testPackageSearch(@NugetRepository(storageId = STORAGE_ID,
...
                .when()
                .get(url, storageId, repositoryId, 0, 30, "Test.Search")
                .then()
                .statusCode(HttpStatus.OK.value())
-               .and()
-               .assertThat()
-               .body("feed.title", equalTo("Packages"))
-               .and()
-               .assertThat()
-               .body("feed.entry[0].title", equalTo(coordinates.getId()));
```
Comment (sbespalov) [inline strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/layout/nuget/NugetArtifactControllerTest.java]: @savaliyagautam what is the purpose of this change?

Comment (savaliyagautam) [reply strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/layout/nuget/NugetArtifactControllerTest.java]: @sbespalo There are 2 parts of it:
**Asserting separately rather than with body method**
- Using rest assure body assert was giving error:
`Caused by: java.lang.ClassNotFoundException: io.restassured.assertion.XMLAssertion`
Debugged to fix the above error, but couldn't found the concrete reason for this error. 

More or less, it is doing same test, which was there, but in 2 steps.

**Changed xpath from `feed.title` to `atom:feed.title`**
Response XML was containing <atom:feed ....>, previously it was testing against only feed xpath.

Comment (sbespalov) [reply strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/layout/nuget/NugetArtifactControllerTest.java]: thanks for clarifying. Can you please check that following dependency exists:
```
<dependency>
  <groupId>io.rest-assured</groupId>
  <artifactId>xml-path</artifactId>
</dependency>
```

Comment (sbespalov) [reply strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/layout/nuget/NugetArtifactControllerTest.java]: I mean it could be gone with new Spring MVC version

Comment (savaliyagautam) [reply strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/layout/nuget/NugetArtifactControllerTest.java]: > thanks for clarifying. Can you please check that following dependency exists:
> 
> ```
> <dependency>
>   <groupId>io.rest-assured</groupId>
>   <artifactId>xml-path</artifactId>
> </dependency>
> ```
@sbespalov  Yes, both xml-path and json-path dependency are there. I tried using `io.restassured.response.ValidatableResponseOptions.body(String, Matcher<?>, Object...)` but as I mentioned it was throwing `Caused by: java.lang.ClassNotFoundException: io.restassured.assertion.XMLAssertion`

Comment (sbespalov) [reply strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/layout/nuget/NugetArtifactControllerTest.java]: @schneems it works for me following way:
```
        <dependency>
            <groupId>io.rest-assured</groupId>
            <artifactId>spring-mock-mvc</artifactId>
            <exclusions>
                <exclusion>
                    <groupId>io.rest-assured</groupId>
                    <artifactId>rest-assured</artifactId>
                </exclusion>
            </exclusions>
            <scope>test</scope>
        </dependency>
        <dependency>
            <groupId>io.rest-assured</groupId>
            <artifactId>rest-assured</artifactId>
            <version>3.1.1</version>
        </dependency>
        <dependency>
            <groupId>io.rest-assured</groupId>
            <artifactId>rest-assured-common</artifactId>
            <version>3.1.1</version>
        </dependency>
        <dependency>
            <groupId>io.rest-assured</groupId>
            <artifactId>json-path</artifactId>
            <version>3.1.1</version>
        </dependency>
        <dependency>
            <groupId>io.rest-assured</groupId>
            <artifactId>xml-path</artifactId>
            <version>3.1.1</version>
        </dependency>
```

can you please make appropriate changes in POMs?

Comment (savaliyagautam) [reply strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/layout/nuget/NugetArtifactControllerTest.java]: @sbespalov 

Upgrading spring.boot from 2.1.5.RELEASE to 2.2.4.RELEASE in https://github.com/strongbox/strongbox-parent/blob/master/pom.xml

is giving below error:
`Caused by: java.lang.ClassNotFoundException: io.restassured.assertion.XMLAssertion`

**Root cause:**

Before upgrade (when spring.boot version - 2.1.5.RELEASE)
```
+- io.rest-assured:spring-mock-mvc:jar:3.0.3:test [version 3.0.3 defined in strongbox parent pom and included under dependencyManagement]
|  +- io.rest-assured:rest-assured:jar:3.0.3:test [version 3.0.3 defined in strongbox parent pom and included under dependencyManagement]
|  |  +- io.rest-assured:xml-path:jar:3.1.1:test [transitive by spring-boot-dependencies/2.1.5.RELEASE]
```

After upgrade (when spring.boot version - 2.2.4.RELEASE)
```
+- io.rest-assured:spring-mock-mvc:jar:3.0.3:test [version 3.0.3 defined in strongbox parent pom and included under dependencyManagement]
|  +- io.rest-assured:rest-assured:jar:3.0.3:test [version 3.0.3 defined in strongbox parent pom and included under dependencyManagement]
|  |  +- io.rest-assured:xml-path:jar:3.3.0:test [transitive by spring-boot-dependencies/2.2.4.RELEASE]
```

**NOTE:** xml-path is getting updated to 3.3.0 from 3.1.1 (as it transitive by spring-boot-dependencies), while rest-assured and spring-mock-mvc is still 3.0.3 and looks like this version mismatch is creating combability issues  between xml-path and rest-assured and throwing error

**Possible solution:**
1. Remove rest-assured from strongbox parent dependencyManagement and let spring-boot-dependencies handle rest-assured version.
2. Update version.rest.assured in strongbox parent from 3.0.3 to 3.3.0 and both rest-assured and xml-path version will be in sync.
3. Add xml-path in strongbox parent dependencyManagement and set version to 3.1.1.
 
@sbespalov  what's your suggestion on this?

Comment (sbespalov) [reply strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/layout/nuget/NugetArtifactControllerTest.java]: the best probably would be №1, but as you noticed this won't suit because we expect restassured update with @anki2189 fix.

> Update version.rest.assured in strongbox parent from 3.0.3 to 3.3.0 and both rest-assured and xml-path version will be in sync.

this won't work for me when I've tried it in your branch, it works only with explicitly `xml-path` dependency with proper version.

so I suggest you to double check №2 and if it won't work (as for me) then use №3.

Comment (sbespalov) [reply strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/layout/nuget/NugetArtifactControllerTest.java]: BTW, this might also be related with some broken artifacts in our Nexus. @steve-todorov can you please also cleanup all restassured related artifacts there?

Comment (savaliyagautam) [reply strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/layout/nuget/NugetArtifactControllerTest.java]: @sbespalov  Applied No.2 (Updated version.rest.assured in strongbox parent from 3.0.3 to 3.3.0)

**Comment by steve-todorov** [comment]
@savaliyagautam I see you have `merged` the master into your branch. Could you please `rebase` instead? 

Merging introduces a lot of noise in the git history and sometime it even messes things up. We try to avoid it. You can check out our [Rebase VS Merging](https://strongbox.github.io/developer-guide/git/rebase-vs-merge.html) guide which goes into more depth.

File: strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/layout/nuget/NugetArtifactControllerTest.java (hunk 1 above)
Comment (steve-todorov) [reply strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/layout/nuget/NugetArtifactControllerTest.java]: @sbespalov I've removed all cached `io.rest-assured` artifacts

**Comment by steve-todorov** [comment]
Thanks for your work @savaliyagautam and @sbespalov for reviewing! :)
